FirebaseAuth/Sources/Backend/FIRAuthBackend.m

```diff
@@ -798,15 +800,25 @@ - (void)verifyAssertion:(FIRVerifyAssertionRequest *)request
                } else {
                  if (!response.IDToken && response.MFAInfo) {
 #if TARGET_OS_IOS
-                   NSMutableArray<FIRMultiFactorInfo *> *multiFactorInfo = [NSMutableArray array];
+                   NSMutableArray<FIRMultiFactorInfo *> *multiFactorInfoArray =
+                       [[NSMutableArray alloc] init];
                    for (FIRAuthProtoMFAEnrollment *MFAEnrollment in response.MFAInfo) {
-                     FIRPhoneMultiFactorInfo *info =
-                         [[FIRPhoneMultiFactorInfo alloc] initWithProto:MFAEnrollment];
-                     [multiFactorInfo addObject:info];
+                     if (MFAEnrollment.phoneInfo) {
+                       FIRMultiFactorInfo *multiFactorInfo =
+                           [[FIRPhoneMultiFactorInfo alloc] initWithProto:MFAEnrollment];
+                       [multiFactorInfoArray addObject:multiFactorInfo];
+                     } else if (MFAEnrollment.TOTPInfo) {
+                       FIRMultiFactorInfo *multiFactorInfo =
+                           [[FIRTOTPMultiFactorInfo alloc] initWithProto:MFAEnrollment];
+                       [multiFactorInfoArray addObject:multiFactorInfo];
+                     } else {
+                       FIRLogError(kFIRLoggerAuth, @"I-AUT000001",
```

Comment:
The string should be unique so that users can reliably Google search log messsages.

Reply:
Got it, thank you Paul!